services/hype-backend/src/modules/reward/reward.service.ts

```diff
@@ -27,16 +36,52 @@ export interface ClaimResult {
 @Injectable()
 export class RewardService {
   privateKey: Buffer;
+
+  gsSecret: string;
+
   private logger = new Logger(RewardService.name);
+
   constructor(
     @Inject(ethereum.KEY)
     private readonly ethereumConfig: ConfigType<typeof ethereum>,
+    @Inject(auth.KEY)
+    private readonly authConfig: ConfigType<typeof auth>,
     @InjectRepository(HypeReward)
     private readonly rewardRepository: Repository<HypeReward>,
     @InjectRepository(HypeClaim)
     private readonly claimRepository: Repository<HypeClaim>,
+    private userService: UsersService,
+    private graphQlService: GraphQlService,
   ) {
     this.privateKey = Buffer.from(this.ethereumConfig.privateSigningKey, 'hex');
+    this.gsSecret = authConfig.gsSecret;
+  }
+
+  @Cron(CronExpression.EVERY_DAY_AT_1AM)
+  async checkClaims() {
+    this.logger.debug('Called every day at 1 AM');
+    const claims = await this.claimRepository.find({
+      where: {
+        claimed: false,
+      },
+    });
+    if (claims.length > 0) {
+      await Promise.all(
+        claims.map(async (claim: HypeClaim) => {
+          const onChainclaims = await this.graphQlService.getClaimedEvents(
+            claim.poolId,
+            claim.rewardee,
+            claim.amount,
+          );
+          if (onChainclaims.claimedEvents.length > 0) {
+            this.logger.log(`Found unclaimed claims`);
```

Comment:
I'd mark this as warn
```suggestion
            this.logger.warn(`Found unclaimed claims`);
```